syscall/syscall.c

```diff
@@ -33,851 +34,1043 @@ typedef int (*ioctl_proc)(
-            goto done;
-        }
-        case OE_SYS_close:
-        {
-            int fd = (int)arg1;
+OE_DEFINE_SYSCALL3(SYS_dup3)
+{
+    oe_errno = 0;
+    long ret = -1;
+    int oldfd = (int)arg1;
+    int newfd = (int)arg2;
+    int flags = (int)arg3;
 
-            ret = oe_close(fd);
-            goto done;
-        }
-        case OE_SYS_dup:
-        {
-            int fd = (int)arg1;
+    if (flags != 0)
+    {
+        oe_errno = OE_EINVAL;
+        goto done;
+    }
 
-            ret = oe_dup(fd);
-            goto done;
-        }
-        case OE_SYS_flock:
-        {
-            int fd = (int)arg1;
-            int operation = (int)arg2;
+    ret = oe_dup2(oldfd, newfd);
+done:
+    return ret;
+}
 
-            ret = oe_flock(fd, operation);
-            goto done;
-        }
-        case OE_SYS_fsync:
-        {
-            const int fd = (int)arg1;
+#if defined(OE_SYS_epoll_create)
+OE_DEFINE_SYSCALL1(SYS_epoll_create)
+{
+    oe_errno = 0;
+    int size = (int)arg1;
+    return oe_epoll_create(size);
+}
+#endif
 
-            ret = oe_fsync(fd);
-            goto done;
-        }
-        case OE_SYS_fdatasync:
-        {
-            const int fd = (int)arg1;
+OE_DEFINE_SYSCALL1(SYS_epoll_create1)
+{
+    oe_errno = 0;
+    int flags = (int)arg1;
+    return oe_epoll_create1(flags);
+}
 
-            ret = oe_fdatasync(fd);
-            goto done;
-        }
-#if defined(OE_SYS_dup2)
-        case OE_SYS_dup2:
-        {
-            int oldfd = (int)arg1;
-            int newfd = (int)arg2;
+OE_DEFINE_SYSCALL4(SYS_epoll_ctl)
+{
+    oe_errno = 0;
+    int epfd = (int)arg1;
+    int op = (int)arg2;
+    int fd = (int)arg3;
+    struct oe_epoll_event* event = (struct oe_epoll_event*)arg4;
+    return oe_epoll_ctl(epfd, op, fd, event);
+}
 
-            ret = oe_dup2(oldfd, newfd);
-            goto done;
-        }
+OE_DEFINE_SYSCALL5(SYS_epoll_pwait)
+{
+    oe_errno = 0;
+    int epfd = (int)arg1;
+    struct oe_epoll_event* events = (struct oe_epoll_event*)arg2;
+    int maxevents = (int)arg3;
+    int timeout = (int)arg4;
+    const oe_sigset_t* sigmask = (const oe_sigset_t*)arg5;
+    return oe_epoll_pwait(epfd, events, maxevents, timeout, sigmask);
+}
+
+#if defined(OE_SYS_epoll_wait)
+OE_DEFINE_SYSCALL4(SYS_epoll_wait)
+{
+    oe_errno = 0;
+    int epfd = (int)arg1;
+    struct oe_epoll_event* events = (struct oe_epoll_event*)arg2;
+    int maxevents = (int)arg3;
+    int timeout = (int)arg4;
+    return oe_epoll_wait(epfd, events, maxevents, timeout);
+}
 #endif
-        case OE_SYS_dup3:
-        {
-            int oldfd = (int)arg1;
-            int newfd = (int)arg2;
-            int flags = (int)arg3;
 
-            if (flags != 0)
-            {
-                oe_errno = OE_EINVAL;
-                goto done;
-            }
+OE_DEFINE_SYSCALL1(SYS_exit)
+{
+    oe_errno = 0;
+    int status = (int)arg1;
+    oe_exit(status);
+    // TODO: Is this correct return value.
```

Comment:
I checked the musl and glibc implementations, and they just don't use the return code of this syscall, so ultimately it probably doesn't matter. oe_exit doesn't return anyway.

Reply:
Good point.